index.js

```diff
@@ -229,26 +227,32 @@ function solveChallenge (options, response, body) {
 
   payload.jschl_vc = match[1];
 
-  match = body.match(/getElementById\('cf-content'\)[\s\S]+?setTimeout.+?\r?\n([\s\S]+?a\.value =.+?)\r?\n/i);
+  match = body.match(/getElementById\('cf-content'\)[\s\S]+?setTimeout.+?\r?\n([\s\S]+?a\.value\s*=.+?)\r?\n/);
 
   if (!match) {
     cause = 'setTimeout callback extraction failed';
     return callback(new errors.ParserError(cause, options, response));
   }
 
   challenge = match[1]
-    .replace(/a\.value =(.+?) \+ .+?;/i, '$1')
-    .replace(/\s{3,}[a-z](?: = |\.).+/g, '')
+    .replace(/a\.value\s*=\s*(.*)/, function (_, value) {
+      return value.replace(/[A-Za-z0-9_$]+\.length/, uri.hostname.length);
+    })
+    .replace(/^\s*[a-z0-9_$]+\s*[=.].+/gm, '')
```

Comment:
@codemanki, I meant to make that case insensitive. It should be `.replace(/^\s*[a-z0-9_$]+\s*[=.].+/gmi, '')` but it mostly doesn't matter. What does matter is that these replacements can fail silently. If they fail silently, you'll get a Cloudflare challenge loop. I think we should update this to throw a ParserError if replacements fail because that means the challenge has changed and we need to update ASAP. Whereas the challenge loop doesn't clearly identify the cause.